Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -240,6 +240,10 @@ open class RouteController: NSObject {
         let legIndex = UInt32(value)
         if navigator.changeRouteLeg(forRoute: 0, leg: legIndex), let status = navigator.getStatus() {
             updateIndexes(status: status, progress: routeProgress)
+            /** NOTE:
+             `navigator.changeRouteLeg(forRoute:leg:)` will return true if the leg actually changed.
+             */
+            BillingHandler.shared.beginNewBillingSessionIfRunning(with: sessionUUID)
```

Comment:
This is the most reliable way I found to identify a new Waypoint start. @1ec5 This is the most critical section of the PR.

Reply:
Yes, I agree that this is the most natural bottleneck for the start of a new leg.